include/aws/common/common.h

```diff
@@ -119,26 +120,46 @@
-#define AWS_PRECONDITION_1(cond) __CPROVER_precondition((cond), #cond " check failed")
-#define AWS_POSTCONDITION_2(cond, explanation) __CPROVER_assert((cond), (explanation))
-#define AWS_POSTCONDITION_1(cond) __CPROVER_assert((cond), #cond " check failed")
-#define AWS_MEM_IS_READABLE(base, len) __CPROVER_r_ok((base), (len))
-#define AWS_MEM_IS_WRITABLE(base, len) __CPROVER_w_ok((base), (len))
+#    define AWS_PRECONDITION_2(cond, explanation) __CPROVER_precondition((cond), (explanation))
+#    define AWS_PRECONDITION_1(cond) __CPROVER_precondition((cond), #cond " check failed")
+#    define AWS_POSTCONDITION_2(cond, explanation) __CPROVER_assert((cond), (explanation))
+#    define AWS_POSTCONDITION_1(cond) __CPROVER_assert((cond), #cond " check failed")
+#    define AWS_MEM_IS_READABLE(base, len) __CPROVER_r_ok((base), (len))
+#    define AWS_MEM_IS_WRITABLE(base, len) __CPROVER_w_ok((base), (len))
 #else
-#define AWS_PRECONDITION_2(cond, expl) AWS_ASSERT(cond)
-#define AWS_PRECONDITION_1(cond) AWS_ASSERT(cond)
-#define AWS_POSTCONDITION_2(cond, expl) AWS_ASSERT(cond)
-#define AWS_POSTCONDITION_1(cond) AWS_ASSERT(cond)
+#    define AWS_PRECONDITION_2(cond, expl) AWS_ASSERT(cond)
+#    define AWS_PRECONDITION_1(cond) AWS_ASSERT(cond)
+#    define AWS_POSTCONDITION_2(cond, expl) AWS_ASSERT(cond)
+#    define AWS_POSTCONDITION_1(cond) AWS_ASSERT(cond)
 /* the C runtime does not give a way to check these properties,
  * but we can at least check that the pointer is valid */
-#define AWS_MEM_IS_READABLE(base, len) (((len) == 0) || (base))
-#define AWS_MEM_IS_WRITABLE(base, len) (((len) == 0) || (base))
-#endif
-
-// The UNUSED is used to silence the complains of GCC for zero arguments in variadic macro
-#define AWS_PRECONDITION(...) GET_MACRO(__VA_ARGS__, AWS_PRECONDITION_2, AWS_PRECONDITION_1, UNUSED)(__VA_ARGS__)
-#define AWS_POSTCONDITION(...) GET_MACRO(__VA_ARGS__, AWS_POSTCONDITION_2, AWS_POSTCONDITION_1, UNUSED)(__VA_ARGS__)
+#    define AWS_MEM_IS_READABLE(base, len) (((len) == 0) || (base))
+#    define AWS_MEM_IS_WRITABLE(base, len) (((len) == 0) || (base))
+#endif /* CBMC */
+
+#define AWS_ERROR_PRECONDITION_2(cond, err)	\
+  do {						\
+    if(!(cond)) {				\
+      return aws_raise_error(err);		\
+    }						\
+  } while(0)
+#define AWS_ERROR_PRECONDITION_3(cond, err, explanation) AWS_ERROR_PRECONDITION_2(cond, err)
+#define AWS_ERROR_PRECONDITION_1(cond) AWS_ERROR_PRECONDITION_2(cond, AWS_ERROR_INVALID_ARGUMENT)
+
+  
+#define AWS_ERROR_POSTCONDITION_2(cond, err)	\
+  do {						\
+    if(!(cond)) {				\
+      return aws_raise_error(err);		\
+    }						\
+  } while(0)
+#define AWS_ERROR_POSTCONDITION_3(cond, err, explanation) AWS_ERROR_POSTCONDITION_2(cond, err)
+
+/* The UNUSED is used to silence the complains of GCC for zero arguments in variadic macro */
+#define AWS_PRECONDITION(...) GET_MACRO(__VA_ARGS__, INVALID, AWS_PRECONDITION_2, AWS_PRECONDITION_1, UNUSED)(__VA_ARGS__)
```

Comment:
Minor nit, use `UNUSED` both places instead of `INVALID`?

Reply:
Done.